torch/_inductor/compile_fx.py

```diff
@@ -219,88 +225,649 @@ def remove_unaligned_input_idxs(inputs, static_input_idxs):
+            "data_ptr": x.storage().data_ptr(),
+            "size": x.shape,
+            "stride": x.stride(),
+            "dtype": x.dtype,
+            "device": x.device,
+            "storage_offset": x.storage_offset() if not ignore_storage_offset else 0,
+        }
 
-    inps_expanded_dims = [
-        get_expanded_dims(x) if idx not in static_input_idxs else []
-        for idx, x in enumerate(inputs)
-    ]
+    @staticmethod
+    def reconstruct_from_tensor_metadata(metadata):
+        s = torch._C._construct_storage_from_data_pointer(
+            metadata["data_ptr"], metadata["device"], metadata["nbytes"]
+        )
+        t = torch.empty([0], device=metadata["device"], dtype=metadata["dtype"])
+        t.set_(
+            source=s,
+            storage_offset=metadata["storage_offset"],
+            size=metadata["size"],
+            stride=metadata["stride"],
+        )
+        return t
 
-    # warmup
-    torch.cuda.synchronize()
-    stream = torch.cuda.Stream()
-    stream.wait_stream(torch.cuda.current_stream())
-    # copy static_inputs because it will be cleared in model
-    with torch.cuda.stream(stream):
-        model(list(static_inputs))
-    stream.synchronize()
-    torch.cuda.current_stream().wait_stream(stream)
-    torch.cuda.synchronize()
-
-    # record
-    graph = torch.cuda.CUDAGraph()
-    with torch.cuda.graph(graph, stream=stream):
-        static_outputs = model(list(static_inputs))
-    if not isinstance(static_outputs, (list, tuple)):
-        static_outputs = (static_outputs,)
-
-    if config.size_asserts:
-
-        def run(new_inputs):
-            assert len(static_inputs) == len(new_inputs)
-            for idx, (dst, src, expanded_dims) in enumerate(
-                zip(static_inputs, new_inputs, inps_expanded_dims)
-            ):
-                if idx in static_input_idxs:
-                    assert dst.data_ptr() == src.data_ptr()
-                else:
-                    # TODO - could make one single op of multiple slices
-                    # and avoid dispatch.
-                    # Could also pre-index the `dst` tensors
-                    dst = index_expanded_dims(dst, expanded_dims)
-                    src = index_expanded_dims(src, expanded_dims)
-                    dst.copy_(src)
-            new_inputs.clear()
-            graph.replay()
-            return static_outputs
-
-    else:
-        copy_indices = [
-            idx for idx in range(len(static_inputs)) if idx not in static_input_idxs
-        ]
 
-        def run(new_inputs):
-            for idx in copy_indices:
-                src = index_expanded_dims(static_inputs[idx], inps_expanded_dims[idx])
-                dst = index_expanded_dims(new_inputs[idx], inps_expanded_dims[idx])
-                dst.copy_(src)
-            new_inputs.clear()
-            graph.replay()
-            return static_outputs
+def cudagraphify_impl(model, inputs, static_input_idxs=()):
```

Comment:
I'm a little bit unclear; does this function get called every time we run the model, or only once when we cudagraphify